sentry-android-core/src/main/java/io/sentry/android/core/AndroidUserCache.java

```diff
@@ -0,0 +1,90 @@
+package io.sentry.android.core;
+
+import android.content.SharedPreferences;
+import io.sentry.core.IUserCache;
+import io.sentry.core.protocol.User;
+import io.sentry.core.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+// TODO: to think, this could be a SentryOptionsCache that could be expanded, does it make sense?
+
+/** the Android User Cache class. It caches the user data in the Android SharedPreferences */
+public final class AndroidUserCache implements IUserCache {
+
+  // TODO: only these 3 fields make sense to me, otherwise caching a raw json would be better, but I
+  // don't want to do that.
+  private static final String EMAIL = "email";
+  private static final String ID = "id";
+  private static final String USERNAME = "user_name";
+
+  private final @NotNull SharedPreferences sharedPreferences;
+  private final @NotNull SentryAndroidOptions options;
+  private final @Nullable String sentryDeviceId;
+
+  public AndroidUserCache(
+      final @NotNull SentryAndroidOptions options,
+      final @NotNull SharedPreferences sharedPreferences,
+      final @Nullable String sentryDeviceId) {
+    this.options = Objects.requireNonNull(options, "SentryAndroidOptions is required.");
+    this.sharedPreferences =
+        Objects.requireNonNull(sharedPreferences, "SharedPreferences is required.");
+    this.sentryDeviceId = sentryDeviceId;
+  }
+
+  //  TODO: I believe this makes sense only for global hub SDKs, but if not,
+  //   how do we cache multiple users (because we have multiple scopes)?
```

Comment:
to discuss

Reply:
yep, this one is Android-specific, but we can't forget that `sentry-core` is cross-platform.
nothing to do here for now.